tests/ignite/metrics/test_precision_recall_curve.py

```diff
@@ -28,85 +28,101 @@ def test_no_sklearn(mock_no_sklearn):
         pr_curve.compute()
 
 
-def test_precision_recall_curve():
+def test_precision_recall_curve(available_device):
     size = 100
-    np_y_pred = np.random.rand(size, 1)
-    np_y = np.zeros((size,))
-    np_y[size // 2 :] = 1
-    sk_precision, sk_recall, sk_thresholds = precision_recall_curve(np_y, np_y_pred)
+    y_pred = torch.rand(size, 1, dtype=torch.float32, device=available_device)
+    y_true = torch.zeros(size, dtype=torch.float32, device=available_device)
+    y_true[size // 2 :] = 1.0
+    expected_precision, expected_recall, expected_thresholds = precision_recall_curve(
+        y_true.cpu().numpy(), y_pred.cpu().numpy()
+    )
 
-    precision_recall_curve_metric = PrecisionRecallCurve()
-    y_pred = torch.from_numpy(np_y_pred)
-    y = torch.from_numpy(np_y)
+    precision_recall_curve_metric = PrecisionRecallCurve(device=available_device)
+    assert precision_recall_curve_metric._device == torch.device(available_device)
 
-    precision_recall_curve_metric.update((y_pred, y))
+    precision_recall_curve_metric.update((y_pred, y_true))
     precision, recall, thresholds = precision_recall_curve_metric.compute()
-    precision = precision.numpy()
-    recall = recall.numpy()
-    thresholds = thresholds.numpy()
 
-    assert pytest.approx(precision) == sk_precision
-    assert pytest.approx(recall) == sk_recall
+    precision = precision.cpu().numpy()
+    recall = recall.cpu().numpy()
+    thresholds = thresholds.cpu().numpy()
+
+    assert pytest.approx(precision) == expected_precision
+    assert pytest.approx(recall) == expected_recall
     # assert thresholds almost equal, due to numpy->torch->numpy conversion
-    np.testing.assert_array_almost_equal(thresholds, sk_thresholds)
+    np.testing.assert_array_almost_equal(thresholds, expected_thresholds)
```

Comment:
Can we use pytest approx here instead of assert_array_almost_equal ?
Please remove `# assert thresholds almost equal, due to numpy->torch->numpy conversion` line